File: alltestcases/TracelifyTest.java
package com.tracelify;

import java.util.HashMap;
import java.util.Map;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.sql.SQLException;
import java.text.ParseException;
import java.util.NoSuchElementException;
import java.util.ConcurrentModificationException;

public class TracelifyTest {
    private static final String DSN = "http://c482f74735cacf53d2ad256c8429f1f7@54.251.156.151.nip.io:8000/api/d19f7705-48bc-4842-af80-55da7bc29007/events";

    private static void capture(Tracelify sdk, int idx, Exception e) {
        System.out.println("[" + idx + "] Capturing " + e.getClass().getSimpleName() + "...");
        sdk.captureException(e);
    }

    public static void main(String[] args) {
        System.out.println("=== Tracelify Java SDK Test (20 Errors) ===");
        Tracelify sdk = new Tracelify(DSN, "1.0.0");
        Map<String, Object> user = new HashMap<>(); user.put("id", "usr_j_1"); sdk.setUser(user);
        
        try { int x = 10/0; } catch (Exception e) { capture(sdk, 1, e); }
        try { String s = null; s.length(); } catch (Exception e) { capture(sdk, 2, e); }
        try { throw new IllegalArgumentException("Bad"); } catch (Exception e) { capture(sdk, 3, e); }
        try { throw new FileNotFoundException("/null"); } catch (Exception e) { capture(sdk, 4, e); }
        try { int[] a = new int[1]; int b = a[5]; } catch (Exception e) { capture(sdk, 5, e); }
        try { throw new IOException("Net err"); } catch (Exception e) { capture(sdk, 6, e); }
        try { throw new SQLException("DB err"); } catch (Exception e) { capture(sdk, 7, e); }
        try { Object o = 1; String s = (String)o; } catch (Exception e) { capture(sdk, 8, e); }
        try { throw new ParseException("Err", 0); } catch (Exception e) { capture(sdk, 9, e); }
        try { throw new UnsupportedOperationException(""); } catch (Exception e) { capture(sdk, 10, e); }
        try { throw new IllegalStateException(); } catch (Exception e) { capture(sdk, 11, e); }
        try { Integer.parseInt("abc"); } catch (Exception e) { capture(sdk, 12, e); }
        try { throw new SecurityException(); } catch (Exception e) { capture(sdk, 13, e); }
        try { Object[] a = new String[1]; a[0] = 1; } catch (Exception e) { capture(sdk, 14, e); }
        try { "a".substring(5); } catch (Exception e) { capture(sdk, 15, e); }
        try { int[] a = new int[-1]; } catch (Exception e) { capture(sdk, 16, e); }
        try { throw new InterruptedException(); } catch (Exception e) { capture(sdk, 17, e); }
        try { throw new ConcurrentModificationException(); } catch (Exception e) { capture(sdk, 18, e); }
        try { throw new NoSuchElementException(); } catch (Exception e) { capture(sdk, 19, e); }
        try { throw new RuntimeException("Generic app failure"); } catch (Exception e) { capture(sdk, 20, e); }

        System.out.println("Shutting down...");
        try { sdk.shutdown(); } catch (Exception e) {}
        System.out.println("✅ Done");
    }
}

File: alltestcases/TracelifyTest.kt
import com.tracelify.Tracelify
import java.io.FileNotFoundException
import java.io.IOException
import java.lang.IllegalArgumentException
import java.lang.IllegalStateException
import java.lang.IndexOutOfBoundsException
import java.lang.NumberFormatException
import java.lang.UnsupportedOperationException
import java.lang.ClassCastException
import java.lang.RuntimeException
import java.util.NoSuchElementException
import java.util.ConcurrentModificationException
import java.text.ParseException

fun main() {
    val DSN = "http://24910fd46216c29e346fb96e87719a0d@54.251.156.151.nip.io:8000/api/99edd5b2-b7c8-44aa-8f8b-ff3280fa3f28/events"
    println("=== Tracelify Kotlin SDK — Integration Test (20 Errors) ===")
    
    val sdk = Tracelify(DSN, "1.0.0")
    sdk.setUser(mapOf("id" to "usr_kt_01", "name" to "Kotlin Tester"))
    sdk.setTag("env", "production")

    fun capture(idx: Int, e: Throwable) {
        println("[$idx] Capturing ${e.javaClass.simpleName}...")
        sdk.captureException(e)
    }

    try { val s: String? = null; s!!.length } catch (e: Exception) { capture(1, e) }
    try { val x = 10 / 0 } catch (e: Exception) { capture(2, e) }
    try { throw FileNotFoundException("/abc") } catch (e: Exception) { capture(3, e) }
    try { throw IllegalArgumentException("Bad arg") } catch (e: Exception) { capture(4, e) }
    try { throw IllegalStateException("Bad state") } catch (e: Exception) { capture(5, e) }
    try { val arr = arrayOf(1); arr[5] } catch (e: Exception) { capture(6, e) }
    try { throw UnsupportedOperationException("Not supported") } catch (e: Exception) { capture(7, e) }
    try { "abc".toInt() } catch (e: Exception) { capture(8, e) }
    try { val a: Any = "A"; val b = a as Int } catch (e: Exception) { capture(9, e) }
    try { throw NoSuchElementException("No element") } catch (e: Exception) { capture(10, e) }
    try { throw ConcurrentModificationException("Modified") } catch (e: Exception) { capture(11, e) }
    try { throw SecurityException("Denied auth") } catch (e: Exception) { capture(12, e) }
    try { "A".substring(5) } catch (e: Exception) { capture(13, e) }
    try { throw RuntimeException("Generic runtime") } catch (e: Exception) { capture(14, e) }
    try { throw IOException("IO error simulated") } catch (e: Exception) { capture(15, e) }
    try { throw IOException("Socket timeout") } catch (e: Exception) { capture(16, e) }
    try { throw IllegalAccessException("Cannot access property") } catch (e: Exception) { capture(17, e) }
    try { throw ParseException("Unparsable date", 0) } catch (e: Exception) { capture(18, e) }
    try { throw ArrayStoreException("Array type mismatch") } catch (e: Exception) { capture(19, e) }
    try { throw Exception("Generic custom exception") } catch (e: Exception) { capture(20, e) }

    println("\nSignaling Shutdown and awaiting flush...")
    sdk.shutdown()
    println("✅ Done")
}
